snuba/datasets/entities/transactions.py

```diff
@@ -94,43 +90,16 @@
 )
 
 
-class TransactionsQueryStorageSelector(QueryStorageSelector):
-    def __init__(self, mappers: TranslationMappers) -> None:
-        self.__transactions_table = get_writable_storage(StorageKey.TRANSACTIONS)
-        self.__transactions_ro_table = get_storage(StorageKey.TRANSACTIONS_RO)
-        self.__mappers = mappers
-
-    def select_storage(
-        self, query: Query, query_settings: QuerySettings
-    ) -> StorageAndMappers:
-        readonly_referrer = (
-            query_settings.referrer
-            in settings.TRANSACTIONS_DIRECT_TO_READONLY_REFERRERS
-        )
-        use_readonly_storage = readonly_referrer or state.get_config(
-            "enable_transactions_readonly_table", False
-        )
-        storage = (
-            self.__transactions_ro_table
-            if use_readonly_storage
-            else self.__transactions_table
-        )
-        return StorageAndMappers(storage, self.__mappers)
-
-
 class BaseTransactionsEntity(Entity, ABC):
     def __init__(self, custom_mappers: Optional[TranslationMappers] = None) -> None:
         storage = get_writable_storage(StorageKey.TRANSACTIONS)
         schema = storage.get_table_writer().get_schema()
 
         pipeline_builder = SimplePipelineBuilder(
-            query_plan_builder=SelectedStorageQueryPlanBuilder(
-                selector=TransactionsQueryStorageSelector(
-                    mappers=transaction_translator
-                    if custom_mappers is None
-                    else transaction_translator.concat(custom_mappers)
-                )
-            ),
+            query_plan_builder=SingleStorageQueryPlanBuilder(
+                storage=get_storage(StorageKey.TRANSACTIONS),
+                mappers=transaction_translator,
```

Comment:
Actually you need that custom_mappers line? How do you know it's safe to remove?

Reply:
you are right